controllers/kuadrant_controller.go

```diff
@@ -380,7 +383,10 @@ func controlPlaneProviderName() string {
 	return env.GetString("ISTIOOPERATOR_NAME", "istiocontrolplane")
 }
 
-func controlPlaneConfigMapName() string {
+func controlPlaneConfigMapName(isSail bool) string {
```

Comment:
I wonder if this function doesn't belong to `pkg/istio/mesh_config.go`. It looks like the name of the ConfigMap is an implementation detail of each kind of Istio operator.

If this is indeed the direction we want to go, WDYT about something like the following?

```go
func (r *KuadrantReconciler) getIstioConfigObjects(ctx context.Context, logger logr.Logger) ([]common.ConfigWrapper, error) {
	// ...
	if err == nil || apierrors.IsNotFound(err) {
		opw := istio.NewOperatorWrapper(iop)
		configMapName = opw.ConfigMapName()
		configsToUpdate = append(configsToUpdate, opw)
	} else if !apimeta.IsNoMatchError(err) {
		// ...
	} else {
		// ...
		opw := istio.NewSailWrapper(ist)
		configMapName = opw.ConfigMapName()
		configsToUpdate = append(configsToUpdate, opw)
	}
	// ...
}
```

Of course, `OperatorWrapper` and `SailWrapper` structs would implement `ConfigMapName() string` as well (apart from `ConfigWrapper`).

Reply:
Yep that's along the lines of what I was thinking. Unfortunately these changes aren't enough though to solve the install issue, still investigating.